addons/website_forum/models/forum.py

```diff
@@ -967,9 +992,9 @@ def _check_general_rights(self, vals={}):
 
     def _check_karma_rights(self, upvote=None):
         # karma check
-        if upvote and not self.post_id.can_upvote:
+        if upvote and not self.post_id.can_upvote and not self.post_id.already_downvoted:
```

Comment:
still some doubts about these 2 lines...

If you downvoted a question by error, you can not upvote to make it neutral ?
//cc @sbuhl

Reply:
Yes you can. This line avoid the karma check in the backend when you already voted.

Reply:
Why not update can_upvote in this case ?